### DISCUSSION TITLE
Tycho 4: Target Resolution with Plugin-based products

### DISCUSSION THREAD

**Discussion by jgsuess** [eclipse-tycho/tycho#3694] [Q&A]
Hello!

I am trying to create a RAP application that reuses EMF, Epsilon and other MDE components, tentatively titled the [MDE DevOps Workbench](https://gitlab.com/committed-consulting/mde-devops-web-workbench). I have used [pomless](https://github.com/eclipse-tycho/tycho/wiki/Tycho-Pomless) and `4.0.6` to set up the project, but the [packaging of the product fails](https://gitlab.com/committed-consulting/mde-devops-web-workbench/-/jobs/6443643063#L158).

```
26908 [ERROR]   Missing requirement: org.eclipse.rap.demo 1.0.0.qualifier requires 'org.eclipse.equinox.p2.iu; slf4j.simple 0.0.0' but it could not be found
```
[as evident in the build log l.168]((https://gitlab.com/committed-consulting/mde-devops-web-workbench/-/jobs/6443643063#L168))

Looking at the `.m2` directory `slf4j.simple` is missing from both the regular maven structure and the `p2` substructure. I am wondering if this a something akin to [`#1502`](https://github.com/eclipse-tycho/tycho/issues/1052), as my product is plugin-based.

Questions:
1: Has anyone had problems with `v4` and plugin-based product resolution?
2: What is the best way to reduce this to a test-case for a bug hunt in case this is novel?

Background, in case you are still reading. Here are few things I have established:

* Both compilation and plugin assembly succeed.
* The product can be executed using a run configuration and functions as expected.
* The product can be validated from the product editor
* The target can be resolved and contains the `slf4j.simple` bundle

**Comment by jgsuess** [comment]
* I have also reverted to tycho `3.0.5` and also get a build failure during packaging, but it is not as specific.
* I have referenced versions from Orbit in my target platform, assuming the ones in the drop may be corrupt, but see no change.
* I see that the `.api` bundle is in `p2/osgi/bundle`, but the `.simple` bundle is not.
* I have tried to use the `.nop` bundle instead of `.simple`, and see the same behaviour.
* I have tried to force the installation using `<extraRequirements>`, but the installation fails.

I believe what I am really missing is a means to see what tycho's target platform is, and how it is computed. I have not found anything on this.

**Reply by laeubi** [reply]
I think you first must ensure that the bundle is actually there, so if you open your target file and go to the content tab is the bundle there?

Second have you tried a feature based product by just create a feature containing all bundles that are currently in the product?

And can you provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue?

**Comment by jgsuess** [comment]
> if you open your target file and go to the content tab is the bundle there?

Yes, the bundle is there. When I look at the path it points to where PDE stores the target though.

> if you open your target file and go to the content tab is the bundle there?

I will try and see how small I can make it. Certainly have to, as I am blocked otherwise.
